lib/src/builder/generation/declaration_parsing.dart

```diff
@@ -321,18 +333,18 @@ class ParsedDeclarations {
       if (factory.variables.variables.length != 1) {
         error('Factory declarations must be a single variable.',
             getSpan(sourceFile, factory.variables));
-      }
+      } else {
+        final variable = factory.variables.variables.first;
+        final expectedInitializer = '$privateSourcePrefix$factoryName';
 
-      final variable = factory.variables.variables.first;
-      final expectedInitializer = '$privateSourcePrefix$factoryName';
-
-      if (variable.initializer != null && expectedInitializer != variable.initializer.toString()) {
-        error(
-            'Factory variables are stubs for the generated factories, and should not have initializers '
-                'unless initialized with a valid variable name for Dart 2 builder compatibility. '
-                'Should be:\n    $expectedInitializer}',
-            getSpan(sourceFile, variable.initializer)
-        );
+        if ((variable?.initializer?.toString() ?? '') != expectedInitializer) {
+          error(
+              'Factory variables are stubs for the generated factories, and should not have initializers '
+                  'unless initialized with a valid variable name for Dart 2 builder compatibility. '
```

Comment:
This message is leaking some legacy. I think we can drop the "...should not have initializers unless initialized with..." part and just tell them that they "should be initialized with <expected initializer>".